src/docs/sphinx/advancedExamples/validationStudies/hydraulicFracture/kgdViscosityDominated/kgdViscosityDominatedQueries.py

```diff
@@ -0,0 +1,116 @@
+
+    xMin = nodeXCoords[0]
+    xMax = nodeXCoords[-1]
+    yMin = nodeYCoords[0]
+    yMax = nodeYCoords[-1]
+    zMin = nodeZCoords[0]
+    zMax = nodeZCoords[-1]
+
+    nXElem = meshElement.get("nx")
+    nYElem = meshElement.get("ny")
+    nZElem = meshElement.get("nz")
+    nXElem = [float(i) for i in nXElem[1:-1].split(",")]
+    nYElem = [float(i) for i in nYElem[1:-1].split(",")]
+    nZElem = [float(i) for i in nZElem[1:-1].split(",")]  
+    nx = nXElem[0]
+    ny = nYElem[0]
+    nz = nZElem[0]
+    
+    return xMin, xMax, yMin, yMax, zMin, zMax, nx, ny, nz
+
+def SaveTimeHistory( name, array, header ):
+    with open(name, 'w') as filehandle:
+        filehandle.write("%s\n" % header )
+        for row in array:
+            for entry in row:
+                filehandle.write("%10.4g" % entry)
+            filehandle.write("\n")
+
+def main():
+    # Load and process GEOSX results
+    # File path
+    hdf5File1Path = "../../../../../../../inputFiles/hydraulicFracturing/KGD_zeroToughness_output.hdf5"
+    xmlFile1Path = "../../../../../../../inputFiles/hydraulicFracturing/kgdViscosityDominated_base.xml"
+    xmlFile2Path = "../../../../../../../inputFiles/hydraulicFracturing/kgdViscosityDominated_benchmark.xml"
+
+    # Read simulation parameters from XML file
+    xMin, xMax, yMin, yMax, zMin, zMax, nx, ny, nz = getMeshSettings(xmlFile2Path)
+    fracHeight = abs(zMax-zMin)    
+    dy = abs(yMax-yMin)/ny
+    dz = abs(zMax-zMin)/nz
+
+    # Read simulation output from HDF5 file
+    # Global Coordinate of Element Center
+    hf = h5py.File(hdf5File1Path, 'r')
+    xl = hf.get('pressure elementCenter')
+    xl = np.array(xl)
+    xcord = xl[0,:,0]
+    ycord = xl[0,:,1]
+    zcord = xl[0,:,2]
+    tl = hf.get('pressure Time')
+    tl = np.array(tl)
+    # Load pressure
+    fpre = hf.get('pressure')  
+    fpre = np.array(fpre)
+    # Load elementAperture
+    aper = hf.get('elementAperture')  
+    aper = np.array(aper)
+    # Load elementArea
+    area = hf.get('elementArea')  
+    area = np.array(area)
+
+    # Find injection location
+    for i in range(0,len(zcord)):
+        if abs(xcord[i]) < 0.01 and abs(ycord[i]/(dy/2.) - 1) < 0.01 and abs(zcord[i]/(dz/2.) - 1) < 0.01:
+           well_ind = i
+
+    # Compute frac length
+    length = np.zeros([len(tl)]) 
+    farea = np.zeros([len(tl)])    
+    for i in range(0,len(tl)):
+        temp = 0
+        for j in range(0,len(aper[0,:])):
+            if aper[i,j]>1.0e-5:
+               temp += area[i,j]
+        
+        farea[i] = temp
+        length[i] = temp/fracHeight
+
+
+    header   = [['      time', '  pressure', '  aperture', '    length']]
+    timehist = []
+    for i in range(0,len(tl)):
+        time = tl[i]
+        injectionPressure = fpre[i,well_ind]
+        injectionAperture = aper[i,well_ind]
+        fractureLength = length[i]
+        timehist.append([float(time), float(injectionPressure), float(injectionAperture), float(fractureLength)])
```

Comment:
Some of these loops could be easily vectorized, which would make them more readable.

For finding the well index, you could do something like:

```python
r2 = x**2 + y**2    # This would also be easy to find a perf not at the origin
well_ind = np.argmin(r2)

Reply:
Good suggestion